File: app/src/main/java/com/hqq/example/adapter/FullPagerSnapAdapter.java
package com.hqq.example.adapter;

import androidx.recyclerview.widget.GridLayoutManager;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.hqq.example.R;

import java.util.ArrayList;
import java.util.List;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core.app.adapter
 * @FileName :   FullPagerSnapAdapter
 * @Date : 2019/6/18 0018  上午 9:53
 * @Email : qiqiang213@gmail.com
 * @Descrive :
 */
public class FullPagerSnapAdapter extends BaseQuickAdapter<String, BaseViewHolder> {
    private int mScreenWidth;
    int itemCountPerRowOrColumn = 3;
    List<Integer> mIntegers;

    public FullPagerSnapAdapter() {
        super(R.layout.item_full_pager_snap);
        mIntegers = new ArrayList<>();
        mIntegers.add(R.color.colorPrimaryDark);
        mIntegers.add(R.color.colorPrimary);
        mIntegers.add(R.color.colorAccent);
    }

    @Override
    protected void convert(BaseViewHolder helper, String item) {
        mScreenWidth = mContext.getResources().getDisplayMetrics().widthPixels;
        GridLayoutManager.LayoutParams layoutParams = (GridLayoutManager.LayoutParams) helper.itemView.getLayoutParams();
        layoutParams.width = (mScreenWidth) / itemCountPerRowOrColumn;
        layoutParams.height = layoutParams.width * 4 / 3;
        helper.itemView.setLayoutParams(layoutParams);

        helper.itemView.setBackgroundResource(mIntegers.get((int) ((Math.random() * 100) % 3)));


        helper.setText(R.id.tv_item, item);


    }
}

File: app/src/main/java/com/easy/example/adapter/FullPagerSnapAdapter.kt
package com.easy.example.adapter

import android.content.Context
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import com.chad.library.adapter4.BaseQuickAdapter
import com.chad.library.adapter4.viewholder.QuickViewHolder
import com.easy.example.R

/**
 * @Author : huangqiqiang
 * @Package : com.easy.core.app.adapter
 * @FileName :   FullPagerSnapAdapter
 * @Date : 2019/6/18 0018  上午 9:53
 * @Email : qiqiang213@gmail.com
 * @Describe :
 */
class FullPagerSnapAdapter : BaseQuickAdapter<String, QuickViewHolder>() {
    private var mScreenWidth = 0
    var itemCountPerRowOrColumn = 3
    var mIntegers: MutableList<Int>

    init {
        mIntegers = ArrayList()
        mIntegers.add(R.color.colorPrimaryDark)
        mIntegers.add(R.color.colorPrimary)
        mIntegers.add(R.color.colorAccent)
    }

    override fun onBindViewHolder(helper: QuickViewHolder, position: Int, item: String?) {
        mScreenWidth = context.resources.displayMetrics.widthPixels
        val layoutParams = helper.itemView.layoutParams as GridLayoutManager.LayoutParams
        layoutParams.width = mScreenWidth / itemCountPerRowOrColumn
        layoutParams.height = layoutParams.width * 4 / 3
        helper.itemView.layoutParams = layoutParams
        helper.itemView.setBackgroundResource(mIntegers[(Math.random() * 100 % 3).toInt()])
        helper.setText(R.id.tv_item, item)
    }


    override fun onCreateViewHolder(context: Context, parent: ViewGroup, viewType: Int): QuickViewHolder {
        return QuickViewHolder(R.layout.item_full_pager_snap, parent)
    }
}